### ISSUE TITLE
Query crashes the server

### ISSUE THREAD

**Issue by eikeCL** [swilly22/redis-graph#74]
Hi,

I am using the query `Match (n) return n` in the Neo4J Browser to get an overview over the content of the database. Doing this in redis crashes the server.

I have cloned `redis-graph` (Branch master; last commit f6f0589bdc0e498483c4a5168bb7f0cd5d5fd94d) and `redis` today and build them.

```
$ ./redis-server --version
Redis server v=4.0.6 sha=00000000:0 malloc=jemalloc-4.0.3 bits=64 build=f4723bf37d9513ef
```

I was trying the demo (had to add quotes to the string literals). And then I tried the query I know from using Neo4J (it might not be part of the openCipher standard) but I think it shouldn't crash the server:

```
127.0.0.1:6379> GRAPH.QUERY social "CREATE (:person {name: 'roi', age: 32})"
1) 1) ""
2) 1) Labels added: 1
   2) Nodes created: 1
   3) Properties set: 2
   4) "Query internal execution time: 0.099000 milliseconds"
127.0.0.1:6379> GRAPH.QUERY social "MATCH (n:person) return n"
1) 1) "n.age,n.name\x00"
   2) "32.000000,roi"
2) 1) "Query internal execution time: 0.112000 milliseconds"
127.0.0.1:6379> GRAPH.QUERY social "MATCH (n) return n"
```

the query only crashes the server when `social` exists:
```
127.0.0.1:6379> GRAPH.QUERY social "MATCH (n) return n"
1) 1) ""
2) 1) "Query internal execution time: 0.070000 milliseconds"
127.0.0.1:6379> GRAPH.QUERY social "CREATE (:person {name: 'roi', age: 32})"
1) 1) ""
2) 1) Labels added: 1
   2) Nodes created: 1
   3) Properties set: 2
   4) "Query internal execution time: 0.096000 milliseconds"
127.0.0.1:6379> GRAPH.QUERY social "MATCH (n) return n"
```

Output from the server: 

```
$ ./redis-server ../redis.conf
3551:C 18 Dec 15:54:17.900 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
3551:C 18 Dec 15:54:17.900 # Redis version=4.0.6, bits=64, commit=00000000, modified=0, pid=3551, just started
3551:C 18 Dec 15:54:17.900 # Configuration loaded
3551:M 18 Dec 15:54:17.901 * Increased maximum number of open files to 10032 (it was originally set to 1024).
                _._                                                  
           _.-``__ ''-._                                             
      _.-``    `.  `_.  ''-._           Redis 4.0.6 (00000000/0) 64 bit
  .-`` .-```.  ```\/    _.,_ ''-._                                   
 (    '      ,       .-`  | `,    )     Running in standalone mode
 |`-._`-...-` __...-.``-._|'` _.-'|     Port: 6379
 |    `-._   `._    /     _.-'    |     PID: 3551
  `-._    `-._  `-./  _.-'    _.-'                                   
 |`-._`-._    `-.__.-'    _.-'_.-'|                                  
 |    `-._`-._        _.-'_.-'    |           http://redis.io        
  `-._    `-._`-.__.-'_.-'    _.-'                                   
 |`-._`-._    `-.__.-'    _.-'_.-'|                                  
 |    `-._`-._        _.-'_.-'    |                                  
  `-._    `-._`-.__.-'_.-'    _.-'                                   
      `-._    `-.__.-'    _.-'                                       
          `-._        _.-'                                           
              `-.__.-'                                               

3551:M 18 Dec 15:54:17.902 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
3551:M 18 Dec 15:54:17.902 # Server initialized
3551:M 18 Dec 15:54:17.902 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
3551:M 18 Dec 15:54:17.902 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
3551:M 18 Dec 15:54:17.902 * Module 'graph' loaded from /home/eike/redis/redis-module-graph/src/redisgraph.so
3551:M 18 Dec 15:54:17.902 * Ready to accept connections


=== REDIS BUG REPORT START: Cut & paste starting from here ===
3551:M 18 Dec 16:07:29.143 # Redis 4.0.6 crashed by signal: 11
3551:M 18 Dec 16:07:29.143 # Crashed running the instuction at: 0x7f2633cee17e
3551:M 18 Dec 16:07:29.143 # Accessing address: 0xf07000
3551:M 18 Dec 16:07:29.143 # Failed assertion: <no assertion failed> (<no file>:0)

------ STACK TRACE ------
EIP:
/lib/x86_64-linux-gnu/libc.so.6(+0x14e17e)[0x7f2633cee17e]

Backtrace:
./redis-server 127.0.0.1:6379(logStackTrace+0x45)[0x46b495]
./redis-server 127.0.0.1:6379(sigsegvHandler+0xb9)[0x46bc59]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x11390)[0x7f2633f7b390]
/lib/x86_64-linux-gnu/libc.so.6(+0x14e17e)[0x7f2633cee17e]
./redis-server 127.0.0.1:6379(sdsnewlen+0xa4)[0x42fd64]
./redis-server 127.0.0.1:6379[0x43a0c0]
./redis-server 127.0.0.1:6379(addReplyBulkCBuffer+0x28)[0x43b258]
./redis-server 127.0.0.1:6379(RM_ReplyWithStringBuffer+0x1b)[0x4937fb]
/home/eike/redis/redis-module-graph/src/redisgraph.so(ResultSet_Replay+0xf7)[0x7f26317e8b87]

------ INFO OUTPUT ------
# Server
redis_version:4.0.6
redis_git_sha1:00000000
redis_git_dirty:0
redis_build_id:f4723bf37d9513ef
redis_mode:standalone
os:Linux 4.4.0-103-generic x86_64
arch_bits:64
multiplexing_api:epoll
atomicvar_api:atomic-builtin
gcc_version:5.4.0
process_id:3551
run_id:26098d5b1cabb39eb59b53359d4b4d858c409fda
tcp_port:6379
uptime_in_seconds:792
uptime_in_days:0
hz:10
lru_clock:3663809
executable:/home/eike/redis/redis-4.0.6/src/./redis-server
config_file:/home/eike/redis/redis-4.0.6/redis.conf

# Clients
connected_clients:1
client_longest_output_list:0
client_biggest_input_buf:0
blocked_clients:0

# Memory
used_memory:830184
used_memory_human:810.73K
used_memory_rss:10203136
used_memory_rss_human:9.73M
used_memory_peak:830184
used_memory_peak_human:810.73K
used_memory_peak_perc:100.04%
used_memory_overhead:815590
used_memory_startup:765848
used_memory_dataset:14594
used_memory_dataset_perc:22.68%
total_system_memory:16668483584
total_system_memory_human:15.52G
used_memory_lua:37888
used_memory_lua_human:37.00K
maxmemory:0
maxmemory_human:0B
maxmemory_policy:noeviction
mem_fragmentation_ratio:12.29
mem_allocator:jemalloc-4.0.3
active_defrag_running:0
lazyfree_pending_objects:0

# Persistence
loading:0
rdb_changes_since_last_save:1
rdb_bgsave_in_progress:0
rdb_last_save_time:1513612457
rdb_last_bgsave_status:ok
rdb_last_bgsave_time_sec:-1
rdb_current_bgsave_time_sec:-1
rdb_last_cow_size:0
aof_enabled:0
aof_rewrite_in_progress:0
aof_rewrite_scheduled:0
aof_last_rewrite_time_sec:-1
aof_current_rewrite_time_sec:-1
aof_last_bgrewrite_status:ok
aof_last_write_status:ok
aof_last_cow_size:0

# Stats
total_connections_received:1
total_commands_processed:3
instantaneous_ops_per_sec:0
total_net_input_bytes:231
total_net_output_bytes:2964
instantaneous_input_kbps:0.05
instantaneous_output_kbps:0.08
rejected_connections:0
sync_full:0
sync_partial_ok:0
sync_partial_err:0
expired_keys:0
evicted_keys:0
keyspace_hits:0
keyspace_misses:0
pubsub_channels:0
pubsub_patterns:0
latest_fork_usec:0
migrate_cached_sockets:0
slave_expires_tracked_keys:0
active_defrag_hits:0
active_defrag_misses:0
active_defrag_key_hits:0
active_defrag_key_misses:0

# Replication
role:master
connected_slaves:0
master_replid:0caf25f4afe9c6bb694c33ecb9e55176871744e2
master_replid2:0000000000000000000000000000000000000000
master_repl_offset:0
second_repl_offset:-1
repl_backlog_active:0
repl_backlog_size:1048576
repl_backlog_first_byte_offset:0
repl_backlog_histlen:0

# CPU
used_cpu_sys:0.47
used_cpu_user:0.24
used_cpu_sys_children:0.00
used_cpu_user_children:0.00

# Commandstats

# Cluster
cluster_enabled:0

# Keyspace
db0:keys=2,expires=0,avg_ttl=0

------ CLIENT LIST OUTPUT ------
id=2 addr=127.0.0.1:53986 fd=7 name= age=785 idle=0 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=32768 obl=19 oll=0 omem=0 events=r cmd=

------ CURRENT CLIENT INFO ------
id=2 addr=127.0.0.1:53986 fd=7 name= age=785 idle=0 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=32768 obl=19 oll=0 omem=0 events=r cmd=
argv[0]: 'GRAPH.QUERY'
argv[1]: 'social'
argv[2]: '(null)'

------ REGISTERS ------
3551:M 18 Dec 16:07:29.144 # 
RAX:00007f2633221f39 RBX:0000000000000011
RCX:00007f2633221f38 RDX:ffffffffffff1ff8
RDI:00007f263322fec0 RSI:0000000000f06ff7
RBP:ffffffffffffffff RSP:00007fff5ee0aaf8
R8 :00007f2633221f39 R9 :000000000000003c
R10:0000000000000007 R11:0000000000000002
R12:0000000000ef9070 R13:0000000000000004
R14:00007f2633221f28 R15:0000000000000011
RIP:00007f2633cee17e EFL:0000000000010283
CSGSFS:0000000000000033
3551:M 18 Dec 16:07:29.145 # (00007fff5ee0ab07) -> 00007fff5ee0ac60
3551:M 18 Dec 16:07:29.145 # (00007fff5ee0ab06) -> 0000000000000001
3551:M 18 Dec 16:07:29.145 # (00007fff5ee0ab05) -> 0000000000ef8e20
3551:M 18 Dec 16:07:29.145 # (00007fff5ee0ab04) -> 000000000043b258
3551:M 18 Dec 16:07:29.145 # (00007fff5ee0ab03) -> 0000000000ef9070
3551:M 18 Dec 16:07:29.145 # (00007fff5ee0ab02) -> ffffffffffffffff
3551:M 18 Dec 16:07:29.145 # (00007fff5ee0ab01) -> 00007f263335d580
3551:M 18 Dec 16:07:29.145 # (00007fff5ee0ab00) -> 000000000043a0c0
3551:M 18 Dec 16:07:29.145 # (00007fff5ee0aaff) -> 00007fff5ee0abb0
3551:M 18 Dec 16:07:29.145 # (00007fff5ee0aafe) -> 00007fff5ee0aba8
3551:M 18 Dec 16:07:29.145 # (00007fff5ee0aafd) -> 00007f26319fc118
3551:M 18 Dec 16:07:29.145 # (00007fff5ee0aafc) -> 0000000000ef9070
3551:M 18 Dec 16:07:29.145 # (00007fff5ee0aafb) -> ffffffffffffffff
3551:M 18 Dec 16:07:29.145 # (00007fff5ee0aafa) -> 00007f263335d580
3551:M 18 Dec 16:07:29.145 # (00007fff5ee0aaf9) -> 0000000000000001
3551:M 18 Dec 16:07:29.145 # (00007fff5ee0aaf8) -> 000000000042fd64

------ FAST MEMORY TEST ------
3551:M 18 Dec 16:07:29.145 # Bio thread for job type #0 terminated
3551:M 18 Dec 16:07:29.145 # Bio thread for job type #1 terminated
3551:M 18 Dec 16:07:29.145 # Bio thread for job type #2 terminated
*** Preparing to test memory region 758000 (98304 bytes)
*** Preparing to test memory region ee6000 (135168 bytes)
*** Preparing to test memory region 7f26319fc000 (4096 bytes)
*** Preparing to test memory region 7f26319fe000 (8388608 bytes)
*** Preparing to test memory region 7f26321ff000 (8388608 bytes)
*** Preparing to test memory region 7f2632a00000 (10485760 bytes)
*** Preparing to test memory region 7f2633800000 (2097152 bytes)
*** Preparing to test memory region 7f2633f66000 (16384 bytes)
*** Preparing to test memory region 7f2634183000 (16384 bytes)
*** Preparing to test memory region 7f2634890000 (16384 bytes)
*** Preparing to test memory region 7f26348b7000 (8192 bytes)
*** Preparing to test memory region 7f26348bb000 (4096 bytes)
.O.O.O.O.O.O.O.O.O.O.O.O
Fast memory test PASSED, however your memory can still be broken. Please run a memory test for several hours if possible.

------ DUMPING CODE AROUND EIP ------
Symbol: (null) (base: (nil))
Module: /lib/x86_64-linux-gnu/libc.so.6 (base 0x7f2633ba0000)
$ xxd -r -p /tmp/dump.hex /tmp/dump.bin
$ objdump --adjust-vma=(nil) -D -b binary -m i386:x86-64 /tmp/dump.bin
------

=== REDIS BUG REPORT END. Make sure to include from START to END. ===

       Please report the crash by opening an issue on github:

           http://github.com/antirez/redis/issues

  Suspect RAM error? Use redis-server --test-memory to verify it.

Segmentation fault (core dumped)
```

**Comment by swilly22** [comment]
For the time being RedisGraph can't figure out what are the properties/attributes of a collapsed node which isn't labeled.
In the above case
`MATCH (n) return n`
`n` is a "collapsed" node which has no label.

I agree the server shouldn't crash and an informative error should be returned.

**Comment by swilly22** [comment]
@eikeCL did you had to modify the demo to get it running?
Currently the last commit on master is eb17cb371c636c804610ed2275bc9118d7992c51

The demos should run "out of the box", please let me know if you've encountered any trouble running either one of them.

**Comment by espang** [comment]
The crash was my main concern. In general i am using labeled nodes. I have a small project where I use Neo4J and I am looking for an alternative database.

Regarding the demo: I haven't seen quotes around the string properties. I got a parse error when I copied the first demo-query (using redis-cli). Adding single quotes around the string properties solved that. E.g.

```
... "CREATE (:person {name: 'roi', age: 32})"
```

I can paste the parse error tomorrow. Will also check the commit

**Comment by swilly22** [comment]
@eikeCL you can divide the queries executed by the demo into two types:
1. Insertions, which are made by a python library "redisgraph-py" which takes a graph structure and generates "CREATE" queries, therefore the query you've mentioned above is auto-generated by redisgraph-py.
2. Read queries, which are specified under social_queries.py and imdb_queries.py, there you'll find explicit queries.

**Comment by eikeCL** [comment]
I was a bit imprecise. I wasn't speaking about the demo folder with python code. I was talking about the documentation [http://redisgraph.io/#with-redis-cli](url)

I have followed the steps on `redisgraph.io` and there it says to clone `git clone https://github.com/RedisLabsModules/redis-module-graph.git` which is a fork of this repo which doesn't contain the commit you mentioned. I have the latest master of `github.com/RedisLabsModules/redis-module-graph`

**Comment by swilly22** [comment]
@eikeCL No problem,
I'm working on a solution which will allow queries such as:
`GRAPH.QUERY social "MATCH (n) return n"`
to work as in Neo.

Regarding the doc, I'll update them.
Thank you.

**Comment by swilly22** [comment]
Resolved, as part of #75 
When a collapsed entity doesn't have a label, all known attributes from all labelled entities will be used.

For example:
Supposed we've got two types of entities within our graph:
1. Person with attributes name and age
2. Country with attributes name and population
When executing `MATCH (x) RETURN x`
result-set will contain all four attributes foreach entity:
result-set header: 'name', 'age', 'population'.
